iis/datadog_checks/iis/iis.py

```diff
@@ -86,20 +95,49 @@ def collect_sites(self, dd_name, metric_func, counter, counter_values):
             ):
                 continue
 
-            tags = self._get_tags(self.SITE, site_name, is_single_instance)
+            tags = self._get_tags(namespace, site_name, is_single_instance)
             try:
                 metric_func(dd_name, value, tags)
             except Exception as e:
                 self.log.error('Error in metric_func: %s %s %s', dd_name, value, e)
 
             if dd_name == 'iis.uptime':
-                self._report_uptime(self.SITE, value, tags)
+                self._report_uptime(namespace, value, tags)
                 if site_name in remaining_sites:
                     self.log.debug('Removing %r from expected sites', site_name)
                     remaining_sites.remove(site_name)
                 else:
                     self.log.warning('Site %r not in expected sites', site_name)
 
+    def collect_app_pools(self, dd_name, metric_func, counter, counter_values):
```

Comment:
Looks like the is some duplication in `def collect_app_pools()` and  `def collect_sites()`. I think that can be factored, the only differences seems to be the `namespace` and `self._app_pools/self._sites`. WDYT?

Reply:
I did this explicitly so it would be easier to introduce different logic. WDYT?

Reply:
Do you think the logic will diverge in the future for this part of the code ?

Reply:
I tried refactor, the code looks really ugly to retain the nice error messages... I think we should keep it like this.

It reminds me of Dan's story https://overreacted.io/goodbye-clean-code/

> My code traded the ability to change requirements for reduced duplication, and it was not a good trade. For example, we later needed many special cases and behaviors for different handles on different shapes. My abstraction would have to become several times more convoluted to afford that, whereas with the original “messy” version such changes stayed easy as cake.